### PR TITLE
Fix doc example

### PR DESCRIPTION
**Pull request by NielsRogge**
# What does this PR do?

This PR fixes the doc example of `xxxForSequenceClassification` models. I wonder how this test passes currently, cause for me it returned an error as the labels are of shape `(batch_size, num_labels)` but the `problem_type` wasn't set to "multi_label_classification".

### CODE REVIEW

File: src/transformers/utils/doc.py [hunk 1]
```diff
@@ -269,9 +269,10 @@ def _prepare_output_docstrings(output_type, config_class, min_indent=None):
     ```python
     >>> # To train a model on `num_labels` classes, you can pass `num_labels=num_labels` to `.from_pretrained(...)`
     >>> num_labels = len(model.config.id2label)
-    >>> model = {model_class}.from_pretrained("{checkpoint}", num_labels=num_labels)
+    >>> model = {model_class}.from_pretrained(
```
Comment (patrickvonplaten) [inline src/transformers/utils/doc.py:272]: This I think we should revert. Why wouldn't this work?

Comment (ydshieh) [reply src/transformers/utils/doc.py:272]: @patrickvonplaten 

I had a previous conversation with @NielsRogge on Slack.

He was using [celine98/canine-s-finetuned-sst2](https://huggingface.co/celine98/canine-s-finetuned-sst2/blob/main/config.json), which has `"problem_type": "single_label_classification",` set in the config.

Due to setting, in the following block,

https://github.com/huggingface/transformers/blob/d55fcbcc50a27f4f5ad8a5c83786905e55212fa0/src/transformers/models/canine/modeling_canine.py#L1322-L1329

the block for the condition `self.config.problem_type is None` is not run, and it continue to be `single_label_classification`, therefore the output is not compatible with the provided labels (which is to work with multiple labels here).

Comment (ydshieh) [reply src/transformers/utils/doc.py:272]: For Roberta, `cardiffnlp/twitter-roberta-base-emotion` is used, and the `problem_type` is not set in the config. Therefore, the model code is able to set it to `multi_label_classification`.

Comment (patrickvonplaten) [reply src/transformers/utils/doc.py:272]: Happy with "force-passing" `single_label_classification` as a flag here to overwrite default configs

Comment (NielsRogge) [reply src/transformers/utils/doc.py:272]: You mean `multi_label_classification`? It's ok to keep this change?

Comment (ydshieh) [reply src/transformers/utils/doc.py:272]: I think it is in the reversed direction:

- `celine98/canine-s-finetuned-sst2` has config is set to `single_label_classification`)

- but this block in `PT_SEQUENCE_CLASSIFICATION_SAMPLE`
    https://github.com/huggingface/transformers/blob/d77680e1ad6a4b57fa766790383e6b68268bf29b/src/transformers/utils/doc.py#L269-L282

    is meant to be `multi_label_classification`. See `labels = torch.nn.functional.one_hot`.
- That's why @NielsRogge needs to add `problem_type="multi_label_classification"` in the call to `from_pretrained`.

I will wait @NielsRogge joining this discussion, since he knows better the reason behind his change.

Comment (ydshieh) [reply src/transformers/utils/doc.py:272]: Well, he is faster than my response ...

Comment (patrickvonplaten) [reply src/transformers/utils/doc.py:272]: Ah I see sorry yes you're right - ok to keep the change for me then!

**Review by patrickvonplaten — APPROVED** [review]

File: src/transformers/utils/doc.py (hunk 1 above)
Comment (patrickvonplaten) [reply src/transformers/utils/doc.py:272]: Nono all good this makes perfect sense, I misunderstood here.
